File: 00_Velocidade de Processamento/divisores.c
#include <stdio.h>
#include <stdlib.h>
#include <time.h>

int main() {
    int x, i;
    clock_t inicio;
    double tempo;
    int *interacao;
    double *tempos;

    printf("Entre com o Valor: ");
    scanf("%d", &x);

    interacao = (int*)malloc(x * sizeof(int));
    tempos = (double*)malloc(x * sizeof(double));

    inicio = clock();

    for (i = 1; i <= x; i++) {
        interacao[i - 1] = i - 1;
        tempo = (double)(clock() - inicio) / CLOCKS_PER_SEC;
        tempos[i - 1] = tempo;

        if (x % i == 0) {
            printf("%d\n", i);
        }
    }

    FILE *fp;
    fp = fopen("processamento.csv", "w");
    fprintf(fp, "Tempo de Processamento,Interacoes\n");
    for (i = 0; i < x; i++) {
        fprintf(fp, "%.2f,%d\n", tempos[i], interacao[i]);
    }
    fclose(fp);

    free(interacao);
    free(tempos);

    return 0;
}

File: 00_Velocidade de Processamento/divisores.py
# 500 Exercícios Resolvidos com Python
# Thiago Barros
# Exercícios resolvidos com base no Livro - 500 Algoritimos Resolvidos (ANITA LOPES E GUTO GARCIA)
# Algoritimo Numero 215
# Capitulo 4

"""
Entrar com o número e imprimir todos os seus divisores

"""
# Nota: Você pode usar esse loop para calcular o tempo de resposta para cada interação

import time as tm
import pandas as pd
x = int(input("Entre com o Valor: "))
interacao = []
inicio = tm.time()
tempo =[]
inter = 0

for i in range(1, x+1):
    interacao.append(i-1)
    agora = tm.time()
    tempo.append(round(agora-inicio,2))
    if x % i == 0:
        print(i)

curva_interacao = pd.DataFrame({'Tempo de Processamento':tempo, 'Interacoes': interacao})
curva_interacao.to_csv("processamento.csv",index=False)